gemini/reasoning-engine/tutorial_vertex_ai_search_rag_agent.ipynb

```diff
@@ -0,0 +1,823 @@
EpCABCQgAQlIQAISkIAEJCABCbSKgEJjq5rLykpAAhKQgAQkIAEJSEACEpCABCQgAQlIoJkEFBqb2S7WSgISkIAEJCABCUhAAhKQgAQkIAEJSEACrSKg0Niq5rKyEpCABCQgAQlIQAISkIAEJCABCUhAAhJoJgGFxma2i7WSgAQkIAEJSEACEpCABCQgAQlIQAISkECrCCg0tqq5rKwEJCABCUhAAhKQgAQkIAEJSEACEpCABJpJQKGxme1irSQgAQlIQAISkIAEJCABCUhAAhKQgAQk0CoCCo2tai4rKwEJSEACEpCABCQgAQlIQAISkIAEJCCBZhJQaGxmu1grCUhAAhKQgAQkIAEJSEACEpCABCQgAQm0ioBCY6uay8pKQAISkIAEJCABCUhAAhKQgAQkIAEJSKCZBBQam9ku1koCEpCABCQgAQlIQAISkIAEJCABCUhAAq0ioNDYquayshKQgAQkIAEJSEACEpCABCQgAQlIQAISaCYBhcZmtou1koAEJCABCUhAAhKQgAQkIAEJSEACEpBAqwgoNLaquaysBCQgAQlIQAISkIAEJCABCUhAAhKQgASaSWDRZlbLWklAAhKQgAQkMJ0J3H333dP59D13CUhAAhKQgAQkIIGGEFhttdUaUpN2VEOPxna0k7WUgAQkIAEJSEACEpCABCQgAQlIQAISkECjCejR2OjmsXISkIAEJCCB6UnAN8fTs909awlIQAISkIAEJCCBdhPQo7Hd7WftJSABCUhAAhKQgAQkIAEJSEACEpCABCTQCAIKjY1oBishAQlIQAISkIAEJCABCUhAAhKQgAQkIIF2E1BobHf7WXsJSEACEpCABCQgAQlIQAISkIAEJCABCTSCgEJjI5rBSkhAAhKQgAQkIAEJSEACEpCABCQgAQlIoN0EFBrb3X7WXgISkIAEJCABCUhAAhKQgAQkIAEJSEACjSCg0NiIZrASEpCABCQgAQlIQAISkIAEJCABCUhAAhJoNwGFxna3n7WXgAQkIAEJSEACEpCABCQgAQlIQAISkEAjCCg0NqIZrIQEJCABCUhAAhKQgAQkIAEJSEACEpCABNpNQKGx3e1n7SUgAQlIQAISkIAEJCABCUhAAhKQgAQk0AgCCo2NaAYrIQEJSEACEpCABCQgAQlIQAISkIAEJCCBdhNQaGx3+1l7CUhAAhKQgAQkIAEJSEACEpCABCQgAQk0goBCYyOawUpIQAISkIAEJCABCUhAAhKQgAQkIAEJSKDdBBQa291+1l4CEpCABCQgAQlIQAISkIAEJCABCUhAAo0goNDYiGawEhKQgAQkIAEJSEACEpCABCQgAQlIQAISaDcBhcZ2t5+1l4AEJCABCUhAAhKQgAQkIAEJSEACEpBAIwgoNDaiGayEBCQgAQlIQAISkIAEJCABCUhAAhKQgATaTUChsd3tZ+0lIAEJSEACEpCABCQgAQlIQAISkIAEJNAIAgqNjWgGKyEBCUhAAhKQgAQkIAEJSEACEpCABCQggXYTUGhsd/tZewlIQAISkIAEJCABCUhAAhKQgAQkIAEJNIKAQmMjmsFKSEACEpCABCQgAQlIQAISkIAEJCABCUig3QQUGtvdftZeAhKQgAQkIAEJSEACEpCABCQgAQlIQAKNIKDQ2IhmsBISkIAEJCABCUhAAhKQgAQkIAEJSEACEmg3AYXGdreftZeABCQgAQlIQAISkIAEJCABCUhAAhKQQCMIKDQ2ohmshAQkIAEJSEACEpCABCQgAQlIQAISkIAE2k1AobHd7WftJSABCUhAAhKQgAQkIAEJSEACEpCABCTQCAIKjY1oBishAQlIQAISkIAEJCABCUhAAhKQgAQkIIF2E1BobHf7WXsJSEACEpCABCQgAQlIQAISkIAEJCABCTSCgEJjI5rBSkhAAhKQgAQkIAEJSEACEpCABCQgAQlIoN0EFBrb3X7WXgISkIAEJCBaqZuwAAAC4UlEQVQBCUhAAhKQgAQkIAEJSEACjSCg0NiIZrASEpCABCQgAQlIQAISkIAEJCABCUhAAhJoNwGFxna3n7WXgAQkIAEJSEACEpCABCQgAQlIQAISkEAjCCg0NqIZrIQEJCABCUhAAhKQgAQkIAEJSEACEpCABNpNQKGx3e1n7SUgAQlIQAISkIAEJCABCUhAAhKQgAQk0AgCCo2NaAYrIQEJSEACEpCABCQgAQlIQAISkIAEJCCBdhNQaGx3+1l7CUhAAhKQgAQkIAEJSEACEpCABCQgAQk0goBCYyOawUpIQAISkIAEJCABCUhAAhKQgAQkIAEJSKDdBBQa291+1l4CEpCABCQgAQlIQAISkIAEJCABCUhAAo0goNDYiGawEhKQgAQkIAEJSEACEpCABCQgAQlIQAISaDcBhcZ2t5+1l4AEJCABCUhAAhKQgAQkIAEJSEACEpBAIwgoNDaiGayEBCQgAQlIQAISkIAEJCABCUhAAhKQgATaTUChsd3tZ+0lIAEJSEACEpCABCQgAQlIQAISkIAEJNAIAgqNjWgGKyEBCUhAAhKQgAQkIAEJSEACEpCABCQggXYTUGhsd/tZewlIQAISkIAEJCABCUhAAhKQgAQkIAEJNIKAQmMjmsFKSEACEpCABCQgAQlIQAISkIAEJCABCUig3QQUGtvdftZeAhKQgAQkIAEJSEACEpCABCQgAQlIQAKNIKDQ2IhmsBISkIAEJCABCUhAAhKQgAQkIAEJSEACEmg3AYXGdreftZeABCQgAQlIQAISkIAEJCABCUhAAhKQQCMIKDQ2ohmshAQkIAEJSEACEpCABCQgAQlIQAISkIAE2k1AobHd7WftJSABCUhAAhKQgAQkIAEJSEACEpCABCTQCAIKjY1oBishAQlIQAISkIAEJCABCUhAAhKQgAQkIIF2E1BobHf7WXsJSEACEpCABCQgAQlIQAISkIAEJCABCTSCwP8FN7nmt/SlyMcAAAAASUVORK5CYII=\">"
```

Comment:
For this image, can you upload it to the GCS Bucket and link it in? These embedded images can make the notebooks slower to download and render.

https://console.cloud.google.com/storage/browser/github-repo;tab=objects?prefix=&forceOnObjectsSortingFiltering=false

Reply:
Makes sense, what about an `images/` folder within `gemini/reasoning-engine` to keep it closer to the content?

Reply:
Got it, thanks for the tip. I need write access to that bucket and I'll put the image there.

Reply:
Done, thanks!